### ISSUE TITLE
Track exceptions

### ISSUE THREAD

**Issue by ianb** [mozilla/togetherjs#817]
We have a limited number of places where we have callbacks, and we could essentially put a `try {} catch {}` around these with a finite amount of work.  Logging this (see also #815) we could get better debugging information.

The places I can think of:

* `setTimeout` and `setInterval`
* `$.Deferred`
* jQuery handlers generally (some work to patch these exists, in some project somewhere, see ravenjs and/or getsentry)
* Our event handling (though maybe this isn't necessary)
* Sometimes we don't use jQuery events, but use `addEventListener` or something directly
* Integrating with CodeMirror or ACE we use events there

Basically we need to wrap any function that is called by someone that is not tracking-exceptions-aware.  That's not every callback, just a number of categories of callback.  Also any place where we already catch all exceptions, we'd have to add logging (currently we're generally doing `console.warn()`).
